core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/FcJsonRpc.java

```diff
@@ -33,22 +34,66 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.plugin.services.MetricsSystem;
+import org.hyperledger.besu.plugin.services.metrics.Counter;
+import org.hyperledger.besu.plugin.services.metrics.LabelledMetric;
+import org.hyperledger.besu.plugin.services.metrics.OperationTimer;
 
 @JsonRpcService
 public class FcJsonRpc {
+
   public static final int SECP_VALUE = 1;
   public static final int BLS_VALUE = 2;
 
   private static final Logger LOG = LogManager.getLogger();
 
   private final ArtifactSignerProvider fcSigners;
-
-  public FcJsonRpc(final ArtifactSignerProvider fcSigners) {
+  private final Counter secpSigningRequestCounter;
+  private final Counter blsSigningRequestCounter;
+  private final Counter wallestListRequestCounter;
+  private final Counter wallestHasRequestCounter;
+  private final Counter walletSignMessageRequestCounter;
+  private final LabelledMetric<OperationTimer> signingTimer;
+
+  public FcJsonRpc(final ArtifactSignerProvider fcSigners, final MetricsSystem metricsSystem) {
     this.fcSigners = fcSigners;
+    this.secpSigningRequestCounter =
+        metricsSystem.createCounter(
+            Web3SignerMetricCategory.FILECOIN,
+            "secp_signing_request_count",
+            "Number of signing requests made for SECP256k1 keys");
+    this.blsSigningRequestCounter =
+        metricsSystem.createCounter(
+            Web3SignerMetricCategory.FILECOIN,
+            "bls_signing_request_count",
+            "Number of signing requests made for BLS keys");
+    this.wallestListRequestCounter =
+        metricsSystem.createCounter(
+            Web3SignerMetricCategory.FILECOIN,
+            "wallet_list_count",
+            "Number of times a Filecoin.WalletList request has been received");
+    this.wallestHasRequestCounter =
+        metricsSystem.createCounter(
+            Web3SignerMetricCategory.FILECOIN,
+            "wallet_has_count",
+            "Number of times a Filecoin.WalletHas request has been received");
+    this.walletSignMessageRequestCounter =
+        metricsSystem.createCounter(
+            Web3SignerMetricCategory.FILECOIN,
+            "wallet_sign_message_count",
+            "Number of times a Filecoin.WalletSignMessage request has been received");
+
+    this.signingTimer =
+        metricsSystem.createLabelledTimer(
+            Web3SignerMetricCategory.FILECOIN,
+            "wallet_sign_timer",
+            "The duration for a signing operation",
+            "curve");
```

Comment:
nit: type? or keyType? type is what is used on the APIs. not that important though